core/app/models/spree/stock/quantifier.rb

```diff
@@ -23,7 +25,7 @@ def initialize(variant, stock_location = nil)
       #   inventory is not tracked on the variant.
       def total_on_hand
         if @variant.should_track_inventory?
-          stock_items.sum(:count_on_hand)
+          stock_items.map(&:count_on_hand).sum
```

Comment:
I think you can pass `sum` a block, so this could just be `stock_items.sum(&:count_on_hand)`.

Reply:
Yes! Not sure why I did not do that in the first place :D